lib/rspec-steps/hook.rb

```diff
@@ -0,0 +1,34 @@
+require 'rspec-steps/dsl'
+
+module RSpec::Steps
+  class Hook < Struct.new(:type, :kind, :action)
+    def rspec_kind
+      case kind
+      when :each
+        warn_about_promotion(type)
+        :all
+      when :step
+        :each
+      else
+        kind
+      end
+    end
+
+    def warn_about_promotion(scope_name)
+      RSpec::Steps.warnings[
+        "#{scope_name} :each blocks declared for steps are always treated as " +
```

Comment:
`before :step` used to have a particular and important meaning. You can still use it in your own code (mostly to reduce the degree to which we'd have to re-write steps code) but I discovered that e.g. waterpig cannot (because reasons.)

All that said: it's better not to use `before :step` and at some future time we might remove it entirely.

What do you think?